render/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/render/jpql/serializer/impl/JpqlCaseSerializer.kt

```diff
@@ -0,0 +1,34 @@
+package com.linecorp.kotlinjdsl.render.jpql.serializer.impl
+
+import com.linecorp.kotlinjdsl.querymodel.jpql.expression.impl.JpqlCase
+import com.linecorp.kotlinjdsl.render.RenderContext
+import com.linecorp.kotlinjdsl.render.jpql.serializer.JpqlRenderSerializer
+import com.linecorp.kotlinjdsl.render.jpql.serializer.JpqlSerializer
+import com.linecorp.kotlinjdsl.render.jpql.writer.JpqlWriter
+import kotlin.reflect.KClass
+
+class JpqlCaseSerializer : JpqlSerializer<JpqlCase<*>> {
+    override fun handledType(): KClass<JpqlCase<*>> {
+        return JpqlCase::class
+    }
+
+    override fun serialize(part: JpqlCase<*>, writer: JpqlWriter, context: RenderContext) {
+        val delegate = context.getValue(JpqlRenderSerializer)
+
+        writer.write("CASE")
+        writer.writeEach(part.whens.entries, separator = " ") {
+            writer.writeIfAbsent(" ")
+            writer.write("WHEN ")
+            delegate.serialize(it.key, writer, context)
+            writer.write(" THEN ")
+            delegate.serialize(it.value, writer, context)
```

Comment:
What about separating white space from writing strings? 

This is because when you write a string directly in Java, the Java compiler treats it as a constant and allows it to be reused. From this point of view, it would be nice if we could separate whitespace from non-whitespace, rather than strings containing whitespace.

Reply:
I have applied the review.